src/ORM/DataObject.php

```diff
@@ -4211,4 +4237,9 @@ protected function mergeRelatedObject($list, $added, $item)
             $this->mergeRelatedObject($list, $added, $joined);
         }
     }
+
+    private function getCacheKeyComponent(): array
+    {
+        return $this->extend('cacheKeyComponent');
```

Comment:
cacheKeyComponent isn't a great name, maybe put the word 'unique' in there to make it relate to the 'UniqueKeyService'

Reply:
@mfendeksilverstripe - would you change this to `$this->extend(' uniqueKeyComponents');` so that everything matches?

Also I think it's worth adding an example to the docblock of this function:

```
public function uniqueKeyComponents(): array
{
    return [$this->owner->MyColumn, $this->owner->AnotherColumn];
}
```

Reply:
Added a doc block with example, but I don't want to rename the extension point as that would break a lot of existing functionality. Note that this is an existing extension point, not a new one.

Reply:
ok makes sense re naming

should be `return [$this->owner->MyColumn];` though?

Reply:
Actually, my current documentation shows the most common case. Typically, the `cacheKeyComponent()` returns just one key as a string (examples are in Versioned or Fluent). The array of strings actually comes from the `extend` method which calls all extensions and collects them into an array. That's also the reason why the extension point is name as singular.

Reply:
OK, fairly confusing situation, best we document it.  Could we make the docblock this?

```
    /**
     * Extension point to add more cache key components.
     * The framework extend method will return combined values from DataExtension method(s) as an array
     * The method on your DataExtension class should return a single scalar value. For example:
     *
     * public function cacheKeyComponent()
     * {
     *      return (string) $this->owner->MyColumn;
     * }
     *
     * @return array
     */
```

Reply:
Sure, I updated the doc block ;)